tests/modules/saml/src/Controller/ServiceProviderTest.php

```diff
@@ -184,6 +218,436 @@ public function testWithAuthIdWithoutEntity(): void
         $c->discoResponse($request);
     }
 
+    /**
+     * Test Login without Session
+     *
+     * @return void
+     */
+    public function testLoginHandleNoQueryNoSession(): void
```

Comment:
I'll give you an example of how to use dataproviders in the unit tests. Most of the tests have the same structure and are just checking that for given query parameters , that login is called with expected options.